torch/_functorch/aot_autograd.py

```diff
@@ -2988,9 +2982,33 @@ def functional_call(*args, **kwargs):
         bw_compiler = fw_compiler
 
     full_args = []
+    # First, the params
     full_args.extend(params_flat)
+
+    aot_autograd_arg_pos_to_source = None
+    # Then, the params 1:1 mapped sources, if relevant.
+    if hasattr(mod, "_param_name_to_source"):
+        aot_autograd_arg_pos_to_source = []
+        # We now know this came from dynamo, and (1) we care about guards,
+        # so setting up aot_autograd_arg_pos_to_source for downstream dedup guards
+        # can now be done safely. (2) Dynamo logic protects the 1:1 sizing below.
+        for name in params.keys():
+            assert name in mod._param_name_to_source, f"{name} not found."
+            aot_autograd_arg_pos_to_source.append(mod._param_name_to_source[name])
+
+    # Next, the input args
     full_args.extend(args)
 
+    # And now their dynamo equivalent, if there
+    if hasattr(mod, "graph"):
+        if aot_autograd_arg_pos_to_source is None:
+            aot_autograd_arg_pos_to_source = []
+        arg_srcs = [x._dynamo_source for x in mod.graph.nodes if x.op == "placeholder" and hasattr(x, "_dynamo_source")]
+        aot_autograd_arg_pos_to_source.extend(arg_srcs)
+
+    if aot_autograd_arg_pos_to_source:
```

Comment:
`aot_autograd_arg_pos_to_source is None`

If it's empty, then I DO want to know about mismatch